snuba/environment.py

```diff
@@ -35,3 +37,5 @@ def setup_sentry() -> None:
     settings.CLICKHOUSE_PORT,
     client_settings={"readonly": True},
 )
+
+reader: Reader[ClickhouseQuery] = NativeDriverReader(clickhouse_ro)
```

Comment:
This will be reused only on split queries or any other request that runs more than one query. Is this correct ?

Reply:
No, this is a global and instantiated at startup

Reply:
So I guess we will have to ensure thread safety as well when we get to the http reader if that is stateful right ?

Reply:
Yes